src/main/java/io/axoniq/demo/giftcard/api/api.kt

```diff
@@ -22,6 +22,23 @@ data class CancelEvt(val id: String)
                 query = "SELECT COUNT(c) FROM CardSummary c WHERE c.id LIKE CONCAT(:idStartsWith, '%')"))
 data class CardSummary(@Id var id: String, var initialValue: Int, var remainingValue: Int) {
     constructor() : this("", 0, 0)
+	override fun equals(other: Any?): Boolean {
+        if (this === other) {
+            return true
+        }
+        if (javaClass != other?.javaClass) {
+            return false
+        }
+        other as CardSummary
+        if (id == other.id) {
+            return true
+        }
+        return false
+    }
+	
+	 override fun hashCode(): Int {
+        return 31
```

Comment:
Returning a constant from `hashCode()`? You'd want different values to return different hashes. Isn't `id.hashCode()` more appropriate?

Reply:
Hi Bert,

Actually this is the way to handle the general case of Hibernate entity DB generated identifiers.
https://vladmihalcea.com/the-best-way-to-implement-equals-hashcode-and-tostring-with-jpa-and-hibernate/

"The database-generated identifiers are a different story. Because the identifier is assigned by the database during flush-time, the consistency guarantee breaks if we implemented the equals and hashCode based on the identifier just like for assigned identifiers."

As in your case, the card id is provided by the user during creation you can actually use the id.hashCode

Reply:
anyway, i just pushed return Objects.hash(id);

Regards,